src/main/java/com/github/manolo8/darkbot/core/utils/factory/EntityFactory.java

```diff
@@ -0,0 +1,86 @@
+package com.github.manolo8.darkbot.core.utils.factory;
+
+import com.github.manolo8.darkbot.core.entities.Barrier;
+import com.github.manolo8.darkbot.core.entities.BasePoint;
+import com.github.manolo8.darkbot.core.entities.BattleStation;
+import com.github.manolo8.darkbot.core.entities.Box;
+import com.github.manolo8.darkbot.core.entities.Entity;
+import com.github.manolo8.darkbot.core.entities.MapNpc;
+import com.github.manolo8.darkbot.core.entities.NoCloack;
+import com.github.manolo8.darkbot.core.entities.Npc;
+import com.github.manolo8.darkbot.core.entities.Ship;
+import com.github.manolo8.darkbot.core.utils.ByteUtils;
+import org.intellij.lang.annotations.Language;
+
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+//Maybe split them same as classes in darkorbit?
+public enum EntityFactory {
+    UNKNOWN     (Entity::new),
```

Comment:
Make unknown the latest entry. Assume they'll be checked top-to-bottom for the first that matches the filter.
Also, make it be either regex or a predicate, so that ship/npc ones can be checked that way